### DISCUSSION TITLE
Which frameworks do you want more investment in?

### DISCUSSION THREAD

**Discussion by bradystroud** [tinacms/tinacms#5247] [Polls]
### Hi TinaCMS fans 👋

TinaCMS is already the best CMS out there for static sites.
Which frameworks do you want our team to be investing in?

https://tina.io/roadmap/

There are a lot of options
- [Next.js](https://nextjs.org/) (Continue to keep on making the most popular one better)
- [Astro](https://astro.build/)
- [Hugo](https://gohugo.io/)
- [Qwik](https://qwik.dev/)
- [Gatsby](https://www.gatsbyjs.com/) (an old one)

Pick the single most important one.

🦙

**Comment by JeanThirion** [comment]
Next is 🐐

**Comment by mmgolden** [comment]
What about Vue/Nuxt?

**Comment by Nierusan** [comment]
Nuxtjs and svelte

**Comment by Theiaz** [comment]
Missing a broader integration with other frameworks outside the react universe. Why is there no bulletpoint for the vue universe (nuxt especially?)

**Comment by yumin-chen** [comment]
I picked Next.js but I also loved using Gatsby

**Comment by onixmahmoudi** [comment]
React router v7 (remix)?

**Reply by Klrfl** [reply]
Exactly. What Nextjs should've been. But I just deployed a React Router 7 website with Tina and there's nothing specific to the framework that needs changes

**Comment by james913** [comment]
Hello team,

I recently implemented Tina CMS in my Astro project and I must say the integration process has been smooth. The combination of these technologies is amazing.

However, I've encountered an issue when trying to render blog posts in .mdx format that use custom Astro components. Tina CMS is unable to properly render these components in the visual editor, resulting in errors when the richt-text field attempts to process the content of my .mdx files.

Is this a known limitation, or is there additional configuration required to enable proper integration between Tina CMS and custom Astro components in MDX files?

In any way i support the Tina + Astro combination

**Reply by joshbermanssw** [reply]
Hi @james913 - we'll be looking into that and will get back to you ASAP :)

**Comment by wicksipedia** [comment]
Big news!!!

We’re getting ready to ship Astro without React. This is really going to improve the performance for all Astro users.

We spent some time working on visual editing in Astro without React and now have TinaCMS working with Astro data islands.

It’s currently beta and we’d love you to give it a spin.
https://github.com/tinacms/tina-astro-starter/

We’re keen to action your feedback

—- Matt Wicks, Product Owner TinaCMS

**Reply by djmtype** [reply]
With the new starter, do we follow still follow these docs for self hosting? https://tina.io/docs/self-hosted/overview

Otherwise, the new Astro starter is meant to work with TinaCloud based on the sample env file.

**Reply by kulesy** [reply]
Yes, self-hosting works with Astro. The starter ships wired to TinaCloud so it's a manual swap, but nothing in `@tinacms/astro` is TinaCloud specific.

1. https://tina.io/docs/self-hosted/manual-setup/ is the main path. Its "Hosting it outside Next.js" section is the Astro-relevant bit: `TinaNodeBackend` takes a Node request and response, so it won't mount in an Astro `APIRoute`. Either run one Node server for both, or deploy the backend as a standalone function.
2. https://tina.io/docs/self-hosted/querying-data/ for pointing your loaders at the database client. On Astro those live in `src/lib/data.ts` (starter) or `src/lib/tina/data.ts` (`init`).
